File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        total=1
        count=0
        output=[]
        for num in nums :
         if num ==0 :count+=1
         else :total=total*num

        for i in nums:
            if count >= 2:
               output.append(0)

            elif count == 1:
               if i == 0: output.append(total)
               else :output.append(0)
            
            else :output.append(total//i)

        return output

File: Data Structures & Algorithms/counting-bits/submission-1.java
class Solution {
    static int func(int n){
        int count=0;
        while(n>0){
            int bits=n%2;
            if(bits ==1)count++;
            n/=2;
        }
        return count;
    }
    
    public int[] countBits(int n) {
        int[] res=new int[n+1];
        for(int i=0;i<=n;i++){
            int value=func(i);
            res[i]=value;
        }
        return res;
    }
}
